type/bundle/src/main/java/iu/type/bundle/TypeBundleSpi.java

```diff
@@ -170,30 +124,35 @@ false, libs, TYPE_SPI_LAYER, new FilteringClassLoader(IuIterable.iter("edu.iu",
 	}
 
 	@Override
-	public IuComponent createComponent(ModuleLayer parentLayer, ClassLoader parent, BiConsumer<Module, Controller> controllerCallback,
-			InputStream componentArchiveSource, InputStream... providedDependencyArchiveSources) throws IOException {
+	public IuComponent scanComponentEntry(ClassLoader classLoader, ModuleLayer moduleLayer, Path pathEntry)
+			throws IOException, ClassNotFoundException {
 		if (delegate == null)
 			throw new IllegalStateException("closed");
-		return delegate.createComponent(parentLayer, parent, controllerCallback, componentArchiveSource,
-				providedDependencyArchiveSources);
+		return delegate.scanComponentEntry(classLoader, moduleLayer, pathEntry);
 	}
 
 	@Override
-	public IuComponent scanComponentEntry(ClassLoader classLoader, Path pathEntry)
-			throws IOException, ClassNotFoundException {
+	public IuComponent createComponent(ClassLoader parent, ModuleLayer parentLayer,
+			Consumer<Controller> controllerCallback, InputStream componentArchiveSource,
+			InputStream... providedDependencyArchiveSources) {
 		if (delegate == null)
 			throw new IllegalStateException("closed");
-		return delegate.scanComponentEntry(classLoader, pathEntry);
+		return delegate.createComponent(parent, parentLayer, controllerCallback, componentArchiveSource,
+				providedDependencyArchiveSources);
 	}
 
 	@Override
 	public synchronized void close() throws Exception {
 		instance = null;
 
 		final var delegate = this.delegate;
```

Comment:
This may have been explained before, but can you remind me why we have to hold a reference to the `delegate` until the end of `close()`?